mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ResourceClusterActor.java

```diff
@@ -141,6 +143,7 @@ static Props props(final ClusterID clusterID, final Duration heartbeatTimeout, D
     public void preStart() throws Exception {
         super.preStart();
         fetchJobArtifactsToCache();
+        fetchDisabledTaskExecutors();
```

Comment:
how does the current disable TE on attributes get persisted? can these be merged?

Reply:
Great question. There is another set called `activeDisableTaskExecutorsRequests` which stores disable TE requests  by attributes. Although it can be merged with the new set I'm proposing, however they seem to be fulfilling a slightly difference goal. The existing map is for ASG updates, while the new one is for containers migration. While for the existing set we expect very few requests and it's fine to have an expiration time, for the new one we can have hundreds of simultaneous requests and we only want to stored them until an external signal reaches the control plane (so we don't really need the expiration logic at all. 

I believe long-term we should move to a structure more similar to the one I'm proposing anyway (related to the discussion we had today).

That said I'm open to merge existing and new sets.

Reply:
Could we use the same model for external and storage purposes but store them differently internally for processing purposes? This way, we get the best of both worlds. We will expose less surface area and fewer endpoints they need to think about and manage while we get the efficiency of having two different models.